lib/Logger.ts

```diff
@@ -1,35 +1,51 @@
-import _ from 'underscore';
+type Parameters = string | Record<string, unknown> | Array<Record<string, unknown>> | Error;
+type ServerLoggingCallbackOptions = {api_setCookie: boolean; logPacket: string};
+type ServerLoggingCallback = (logger: Logger, options: ServerLoggingCallbackOptions) => Promise<{requestID: string}> | undefined;
+type ClientLoggingCallBack = (message: string) => void;
+type LogLine = {message: string; parameters: Parameters; onlyFlushWithOthers?: boolean; timestamp: Date};
+type LoggerOptions = {serverLoggingCallback: ServerLoggingCallback; isDebug: boolean; clientLoggingCallback: ClientLoggingCallBack};
 
 const MAX_LOG_LINES_BEFORE_FLUSH = 50;
+
 export default class Logger {
-    constructor({serverLoggingCallback, isDebug, clientLoggingCallback}) {
+    logLines: LogLine[];
+
+    serverLoggingCallback: ServerLoggingCallback;
+
+    clientLoggingCallback: ClientLoggingCallBack;
+
+    isDebug: boolean;
+
+    constructor({serverLoggingCallback, isDebug, clientLoggingCallback}: LoggerOptions) {
         // An array of log lines that limits itself to a certain number of entries (deleting the oldest)
         this.logLines = [];
         this.serverLoggingCallback = serverLoggingCallback;
         this.clientLoggingCallback = clientLoggingCallback;
         this.isDebug = isDebug;
 
         // Public Methods
+        // eslint-disable-next-line no-constructor-return
```

Comment:
Curious why we'd need to disable this lint rule? Couldn't we just change this to:

```ts
this.info = this.info.bind(this);
this.alert = this.alert.bind(this);
this.warn = this.warn.bind(this);
this.hmmm = this.hmmm.bind(this);
this.client = this.client.bind(this);
```

?

Reply:
I didn't want to cause regressions with a change like this, but this should be easily testable, gonna do it today.